torch/_dynamo/variables/ctx_manager.py

```diff
@@ -1026,34 +1027,50 @@ def __init__(self, target_values, device, initial_values=None, **kwargs) -> None
         self.set_stream_id = get_interface_for_device(self.device)._set_stream_by_id
 
     def enter(self, tx):
-        # stream generated inside the traced function
-        if self.target_values[0].as_proxy() is not None:
-            tx.output.create_proxy(
-                "call_function",
-                self.set_stream,
-                (self.target_values[0].as_proxy(),),
-                {},
-            )
-        # stream passed from outside the traced function
-        else:
-            stream = self.target_values[0].value
-            tx.output.create_proxy(
-                "call_function",
-                self.set_stream_id,
-                (stream.stream_id, stream.device_index, stream.device_type),
-                {},
-            )
-        self.set_stream(self.target_values[0].value)
-        self.set_cleanup_hook(tx, lambda: self.set_stream(self.initial_values[0].value))
+        stream_proxy = self.target_values[0].as_proxy()
+        stream_id, device, device_index = (
+            StreamContextVariable._extract_stream_properties(stream_proxy)
+        )
+        proxy = tx.output.create_proxy(
+            "call_function",
+            torch.ops.streams.fork.default,
+            (stream_id, device, device_index, []),
+            {},
+        )
+        stream_state_mgr.push_stream_state(proxy.node)
 
     def exit(self, tx: "InstructionTranslator", *args):
-        tx.output.create_proxy(
+        state = stream_state_mgr.pop_stream_state()
```

Comment:
Would appreciate some comments, especially about the state of `fork` being populated later.